src/SprykerMiddleware/Zed/Report/Persistence/Mapper/ProcessResultMapper.php

```diff
@@ -0,0 +1,104 @@
+        $processResultEntityTransfer->setFailedItemCount($processResultTransfer->getFailedItemCount());
+        $processResultEntityTransfer->setSkippedItemCount($processResultTransfer->getSkippedItemCount());
+        $processResultEntityTransfer->setItemCount($processResultTransfer->getItemCount());
+        $processResultEntityTransfer->setStartTime($processResultTransfer->getStartTime());
+        $processResultEntityTransfer->setEndTime($processResultTransfer->getEndTime());
+
+        $stagesResults = $this->mapStagesResultToArray($processResultTransfer->getStageResults());
+        $processResultEntityTransfer->setStageResults($this->utilEncoding->encodeJson($stagesResults));
+
+        $processResultEntityTransfer->setProcessConfiguration($this->utilEncoding->encodeJson($processResultTransfer->getProcessConfiguration()->toArray()));
+
+        return $processResultEntityTransfer;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\SpyProcessResultEntityTransfer $processResultEntityTransfer
+     *
+     * @return \Generated\Shared\Transfer\ProcessResultTransfer
+     */
+    public function mapEntityTransferToProcessResultTransfer(SpyProcessResultEntityTransfer $processResultEntityTransfer): ProcessResultTransfer
+    {
+        $processResultTransfer = new ProcessResultTransfer();
+        $processResultTransfer->setProcessedItemCount($processResultEntityTransfer->getProcessedItemCount());
+        $processResultTransfer->setFailedItemCount($processResultEntityTransfer->getFailedItemCount());
+        $processResultTransfer->setSkippedItemCount($processResultEntityTransfer->getSkippedItemCount());
+        $processResultTransfer->setItemCount($processResultEntityTransfer->getItemCount());
+        $processResultTransfer->setStartTime($processResultEntityTransfer->getStartTime());
+        $processResultTransfer->setEndTime($processResultEntityTransfer->getEndTime());
+
+        $stages = $this->mapArrayToStagesResult($this->utilEncoding->decodeJson($processResultEntityTransfer->getStageResults(), true));
+        $processResultTransfer->setStageResults($stages);
+
+        $conf = new ProcessConfigurationTransfer();
+        $conf->fromArray($this->utilEncoding->decodeJson($processResultEntityTransfer->getProcessConfiguration(), true));
+        $processResultTransfer->setProcessConfiguration($conf);
+
+        return $processResultTransfer;
+    }
+
+    /**
+     * @param \ArrayObject $stagesResults
+     *
+     * @return array
+     */
+    public function mapStagesResultToArray(ArrayObject $stagesResults)
+    {
+        $stagesResultsArray = [];
+
+        foreach ($stagesResults as $stage) {
+            $stagesResultsArray[] = $stage->toArray();
+        }
+        return $stagesResultsArray;
+    }
+
+    /**
+     * @param array $stagesResults
+     *
+     * @return \ArrayObject
+     */
+    public function mapArrayToStagesResult(array $stagesResults)
```

Comment:
Pass $processResultTransfer to this method too, and use $processResultTransfer->addStageResult(stagesResultsTransfer) instead temp variable